File: text_forge_backend/schema/response/memory.py

from pydantic import BaseModel, ConfigDict, Field


class AgentMemoryResponse(BaseModel):
    id: int
    user_id: int = Field(alias="userId")
    book_id: int | None = Field(default=None, alias="bookId")
    memory_type: str = Field(alias="memoryType")
    content: str
    related_chapter_id: int | None = Field(default=None, alias="relatedChapterId")
    related_character_ids: list[int] | None = Field(default=[], alias="relatedCharacterIds")
    priority: int
    source: str
    meta: dict | None = None
    created_at: str | None = Field(default=None, alias="createdAt")
    updated_at: str | None = Field(default=None, alias="updatedAt")
    distance: float | None = None

    model_config = ConfigDict(populate_by_name=True, from_attributes=True)

File: text_forge_front/src/types/memory.ts
export interface AgentMemory {
  id: number
  userId: number
  bookId?: number
  memoryType: 'preference' | 'rule' | 'fact' | 'user_manual' | 'agent_self_reflection'
  content: string
  relatedChapterId?: number
  relatedCharacterIds: number[]
  priority: number
  source: string
  meta: Record<string, unknown>
  createdAt: string
  updatedAt: string
}